lib/auxiliary/sysinfo_parser.rb

```diff
@@ -0,0 +1,61 @@
+# frozen_string_literal: true
+
+require 'csv'
+require 'active_support'
+require 'active_support/core_ext/object/deep_dup'
+
+# AutoHCK module
+module AutoHCK
+  # SysInfoParser class
+  class SysInfoParser
+    DELIM = ':' # k/v delimiter
+
+    # Main text parsing function
+    #
+    # @param data [String] Text data to parse
+    #
+    # @return [Array<Hash>] List of dictionaries. Processed structured data.
+    def parse(data)
+      rows = CSV.parse(data, headers: true)
+      raw_data = rows.first.to_h
+
+      raw_output = {}
+      raw_data.each do |k, v|
+        raw_output[k] = case k
+                        when 'Hotfix(s)', 'Processor(s)'
+                          parse_hotfixs_or_processors(v)
+                        else
+                          v.strip
+                        end
+      end
+
+      raw_output
+    end
+
+    private
+
+    # Turns a list of return-delimited hotfixes or processors
+    # with [x] as a prefix into an array of hotfixes
+    #
+    # Note that if a high number of items exist, the list may cutoff
+    # and this list may not contain all items installed on the system
+    #
+    # IRL, this likely applies to hotfixes more than processors
+    #
+    # Parameters:
+    #   data: (string) Input string
+    def parse_hotfixs_or_processors(data)
+      arr_output = []
+      data.split(',').each_with_index do |l, i|
+        # skip line that says how many are installed
+        next if i.zero?
+
+        _, v = l.split(DELIM)
+        # discard the number sequence
+        arr_output << v.strip
+      end
+
+      arr_output
```

Comment:
I suggest to write the literal for `:` instead of referring to a constant as you do for `','`. It makes less sense to define a class-wide constant for something only used in this method, and the name is also confusing, considering `,` is also a _delimiter_.

Destructing assignment is used to extract the value, but I think the index syntax is simpler here as you only need an element.

Ruby provides various methods for array manipulation so use them to make it simpler.

You should specify the limit parameter when splitting the index and value so that it won't misbehave when the value includes `:`.

Wrapping up all of the suggestions:

```suggestion
      arr_output = data.split(',')

      # skip line that says how many are installed
      arr_output.shift

      # discard the number sequence
      arr_output.map! { _1.split(':')[1].strip }
```